src/daft-dsl/src/resolve_expr/mod.rs

```diff
@@ -217,19 +217,24 @@ fn expand_wildcards(
 /// - sum(col("a"))
 /// - sum(col("a")) > 0
 /// - sum(col("a")) - sum(col("b")) > sum(col("c"))
+/// - sum(col("a")) + col("b") when "b" is a group by key
 ///
 /// Not allowed:
-/// - col("a")
+/// - col("a") when "a" is not a group by key
 ///     - not an aggregation
-/// - sum(col("a")) + col("b")
-///     - not all branches are aggregations
-fn has_single_agg_layer(expr: &ExprRef) -> bool {
-    match expr.as_ref() {
-        Expr::Agg(agg_expr) => !agg_expr.children().iter().any(has_agg),
-        Expr::Column(_) => false,
-        Expr::Literal(_) => true,
-        _ => expr.children().iter().all(has_single_agg_layer),
-    }
+/// - sum(col("a")) + col("b") when "b" is not a group by key
+///     - not all branches are aggregations, literals, or group by keys
+fn has_single_agg_layer(expr: &ExprRef, groupby: &HashSet<ExprRef>) -> bool {
+    groupby.contains(expr)
+        || match expr.as_ref() {
+            Expr::Agg(agg_expr) => !agg_expr.children().iter().any(has_agg),
+            Expr::Column(_) => false,
+            Expr::Literal(_) => true,
+            _ => expr
+                .children()
+                .iter()
+                .all(|e| has_single_agg_layer(e, groupby)),
+        }
```

Comment:
hmmm I'm not sure this function really tests `has_single_agg_layer` anymore. Maybe rename or rethink logic? or make into two separate functions?

also potential bug (although unsure it could even occur):

groupby contains expr with multiple aggs -> returns true (as in single agg) even though not.

Reply:
Good point but I'm not really sure what to name it. I think the logic of the function itself is fairly straightforward though. I'll give it a thought

As for the bug, group bys should never have aggregations in them and will fail in another place in the code if they do.